### DISCUSSION TITLE
Packages are no longer recognised

### DISCUSSION THREAD

**Discussion by nickmc** [Homebrew/discussions#2721] [Getting started] [answered]
Hi

I just updated to Monterey 12.1 and brew no longer lists all my packages.  I can see the existing packages in "user/local/Cellar" but brew no longer finds them.

"brew list" now only returns:

```
==> Formulae
exa
```

"brew config" returns:

```
HOMEBREW_VERSION: 3.3.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: 96137bc19e68398ebbb7033379df288cd8b9a3f9
Last commit: 2 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 03148b64a605f6467dbc2b9e51202374362613dd
Core tap last commit: 28 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false
```

"brew doctor" does not return any errors. I have reinstalled brew but no change. I'm on a mac m1 and this installation was ported from an earlier mac - but it worked on earlier versions of Monterey.

Is this a problem that anyone else has had? Does anyone have any ideas?

Many thanks
NIck

**Comment by neutric** [comment, marked as answer]
Homebrew on Apple Silicon is in `/opt/homebrew`. What you ported from was on Intel where `HOMEBREW_PREFIX` is `/usr/local`. What you experience is thus expected. Check out the related discussion in https://github.com/Homebrew/discussions/discussions/417. Especially `brew bundle dump` might be useful. It writes all installed casks/formulae/images/taps into a `Brewfile`, which you can then use to reinstall in the new location.

**Reply by nickmc** [reply]
Thanks. Ok that makes sense. I can install new packages in `/opt/homebrew`. I can't do anything with old packages in `/usr/local` (and I can't run `brew bundle dump` on it). Is there a way I can use both? (or even just install the old ones properly?)

**Reply by neutric** [reply]
You probably have `/opt/homebrew/bin` in your PATH before `/usr/local/bin`, which means that `brew` will be the "new" one (=the one for Apple Silicon living in `/opt/homebrew`). You can access the Intel installation by using `/usr/local/bin/brew`, so `/usr/local/bin/brew bundle dump` etc.

**Reply by nickmc** [reply]
OK, many thanks.

**Reply by neutric** [reply]
Welcome!
